lib/events.js

```diff
@@ -182,22 +181,20 @@ api.add = async (
 api.create = async ({ledgerNode, worker}) => {
   logger.verbose('Attempting to create an operation event.');
 
-  const ledgerNodeId = ledgerNode.id;
-  const queue = new _cache.OperationQueue({ledgerNodeId});
+  const queue = worker.operationQueue;
   if(!await queue.hasNextChunk()) {
     logger.debug('No new operations.');
     return {hasMore: false};
   }
 
   // get chunk of operations to put into the event
-  // FIXME: `basisBlockHeight` should be determined by the worker
-  const {basisBlockHeight, hasMore, operations} = await queue.getNextChunk();
+  const {hasMore, operations} = await queue.getNextChunk();
 
   const {head} = worker;
   const baseEvent = {
     '@context': config.constants.WEB_LEDGER_CONTEXT_V1_URL,
     type: 'WebLedgerOperationEvent',
-    basisBlockHeight,
```

Comment:
This wasn't raised, but this is my biggest concern. The previous implementation depended on the queue to receive the proper basis block height. The new behavior assumes that `worker.consensusState.blockHeight` is the same block height that the operations received were validated at. What guarantees that in the worker? Is it the case that the `worker.consensusState.blockHeight` will always match the blockheight the operations were validated at?

Reply:
I thought the queue was setup to run operation validation at a certain block height -- and this block height was provided by the worker. If that's true -- and we don't use left over or "stale" operation validations from a previous block height, we should be good. We should discuss more at some point.